src/main/java/fi/aalto/cs/apluscourses/model/MainViewModelUpdater.java

```diff
@@ -0,0 +1,172 @@
+package fi.aalto.cs.apluscourses.model;
+
+import com.intellij.openapi.application.ReadAction;
+import com.intellij.openapi.project.Project;
+
+import fi.aalto.cs.apluscourses.intellij.model.APlusProject;
+import fi.aalto.cs.apluscourses.intellij.model.IntelliJModelFactory;
+import fi.aalto.cs.apluscourses.intellij.services.MainViewModelProvider;
+import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import fi.aalto.cs.apluscourses.presentation.CourseViewModel;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class MainViewModelUpdater {
```

Comment:
could you please review the access modifiers for the methods inside this class, please?